### PR TITLE
Move existing code into a module

### PR DESCRIPTION
**Pull request by wtimme**
This makes it a bit easier to manage the dependencies and to get new developers set up quickly.
All they need to do now is run `pip install .` and run the module using `python -m generator`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,6 +29,5 @@ None of them is dealing with jira.
 ## Requirements
 - python 2.x
 - [jira-python](https://github.com/pycontribs/jira)
-    - install via `pip install jira`
```
Comment (k3muri84) [inline README.md:22]: we should add that pip needs to be installed

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+from setuptools import setup
+
+setup(name='generate-changelog',
+      version='0.1',
+      description='Generates changelog from Git based on Jira tickets',
+      url='https://github.com/echorebel/generate-changelog',
+      author='echorebel',
+      author_email='echorebel@users.noreply.github.com',
+      license='MIT',
+      packages=['generator'],
```
Comment (k3muri84) [inline setup.py:10]: do we need a more specific package name?

**Comment by k3muri84** [comment]
what do you think about deploying it to build jobs? right now one just need to copy over a file and tweak some props inside it. with this pr we need two files.

**Comment by wtimme** [comment]
Thanks for your feedback! I've added another pull request (#2) after reading up on best practises with Python and dependencies, so I don't think _this_ PR is needed anymore.

**Comment by wtimme** [comment]
I just had a look for your question, since it is also valid for the new pull request:

> what do you think about deploying it to build jobs?

We can [bundle the files into a ZIP file][1]. Then we only need to deploy that ZIP file.

As for the configuration, how about we move that to a file? Do you have experience with Python and storing settings? We could also add support for command line parameters. What do you think?

[1]: http://blog.ablepear.com/2012/10/bundling-python-files-into-stand-alone.html

**Comment by k3muri84** [comment]
hey, it makes sense to move all config to a file, but still i think it's a bit overkill.
one block in the file header should be sufficient and easy to handle.
i'm adding cmdline params to the script - for now just to add build type info but can be extended of course.

**Pull request closed without merging**